azopsreference/3fc1081d-6105-4e19-b60c-1ec1252cf560 (3fc1081d-6105-4e19-b60c-1ec1252cf560)/contoso (contoso)/.AzState/Microsoft.Authorization_policyDefinitions-Deploy-Sql-IaaSExtension-For-AHUB.parameters.json

```diff
@@ -0,0 +1,124 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "input": {
+      "value": {
+        "Name": "Deploy-Sql-IaaSExtension-For-AHUB",
+        "ResourceType": "Microsoft.Authorization/policyDefinitions",
+        "SubscriptionId": null,
+        "Properties": {
+          "Description": "Deploy SQL IaaS Extension to enable SQL Server VMs with the Azure Hybrid Benefit",
+          "DisplayName": "Deploy-Sql-IaaSExtension-For-AHUB",
+          "Mode": "All",
+          "Parameters": { 
+            "tagName": {
+              "type": "String",
+              "metadata": {
+                "displayName": "SQL VM Tag Name",
+                "description": "Name of the tag that you use to identify SQL IaaS VMs to which Azure Hybrid Benefit will be applied, i.e., isSqlVM"
+              }
+            },
+            "sqlManagement": {
+              "type": "String",
+              "allowedValues": [
+                "Lightweight",
+                "Full"
+              ],        
+              "metadata": {
+                "displayName": "SQL Management",
+                "description": "Lightweight, or Full SQL Management. Note: Choosing Full SQL Management would restart the SQL Server in your VM"
+              }
+            }            
+          },
+          "PolicyRule": {
+            "if": {
+              "allOf": [
+                {
+                  "field": "type",
+                  "equals": "Microsoft.Compute/virtualMachines"
+                },
+                {
```

Comment:
using tag to determine if policy should apply or not is non-deterministic and will most often lead to a non-compliant state as tag is not provided upon create request (unless enforced). For the comment above, can we see how we filter and navigate for appropriate images only?

Reply:
Using tag really came about because there is no information within the VM resource ARM template which shows that there is SQL Server installed within it, rightly so. We shouldn't apply this DINE effect on Windows VMs which do not have SQL Server installed because then it leads to a whole bunch of non-compliance VM resources which are not meant to have SqlIaasExtension installed. Any ideas?